build using Meson instead of automake/autoconf

LudovicRousseau opened the pull request.

LudovicRousseau committed:
Initial meson build files

LudovicRousseau committed:
github action: add build_meson

Jakuje commented on meson.build:
```diff
@@ -0,0 +1,260 @@
+  'PCSCLITE_CONFIG_DIR' : get_option('serialconfdir'),
+  'usbdropdir' : get_option('usbdropdir'),
+  })
+features = []
```
nit: I would move this below the compiler checks under the `# tests for options` as it is the first place accessing this array.

LudovicRousseau replied:
done

Jakuje commented on meson.build:
```diff
@@ -0,0 +1,260 @@
+  ['--date=2024-01-01', 'src/spy/pcsc-spy.pod', 'pcsc-spy.1'],
+  check : true)
+install_data('pcsc-spy.1',
+  install_dir : get_option('mandir') + '/man1')
```
nit: `join_paths` is more portable when merging paths than simple concatenation.

LudovicRousseau replied:
Done

Jakuje commented on meson.options:
```diff
@@ -0,0 +1,10 @@
+option('systemdsystemunit', type : 'string', value : '/usr/lib/systemd/system')
+option('serialconfdir', type : 'string', value : '/etc/reader.conf.d')
+option('usbdropdir', type : 'string', value : '/usr/lib/pcsc/drivers')
+option('ipcdir', type : 'string', value : '/run/pcscd')
```
The options support optional `description` key, which I believe would be useful for some human-readable description of the particular option.

LudovicRousseau replied:
Done

Jakuje reviewed (commented):
There are few assorted notes inline and some general in this comment.

The meson support `subdir()` option which can help to separate different parts of 250+ lines long file, avoid repeating the directories (src, doc, etc, ...) and keeping in the main one only the bare minimum. OTOH, it could make harder to navigate the buildsystem if you have to touch more files.

In any case, I am not a meson expert, but I worked with couple of projects already, so if you want some more inspiration, the https://github.com/p11-glue/p11-kit/ is using meson quite successfully for quite some time.

LudovicRousseau committed:
Update c.sh for meson(1)

LudovicRousseau committed:
auth.c: fix compiler warning in embedded mode

../src/auth.c: In function ‘IsClientAuthorized’:
../src/auth.c:83:21: error: unused variable ‘e’ [-Werror=unused-variable]
   83 |                 int e = errno;
      |                     ^

LudovicRousseau committed:
Initial meson build files

LudovicRousseau committed:
github action: add build_meson

LudovicRousseau commented:
The branch is now merged in https://github.com/LudovicRousseau/PCSC
